### PR TITLE
Add support for regular expressions containing hexadecimal digits greater than `0x7f`

### LINKED ISSUE #4866: [BUG] Add support for regular expressions containing hexadecimal digits greater than `0x7f`
**Describe the bug**
We support hexadecimal digits but with limitations. We call back to CPU for certain ranges.

**Expected behavior**
We should support the full range of hexadecimal digits on the GPU (from `Character.MIN_CODE_POINT` to `Character.MAX_CODE_POINT`)

### PR DESCRIPTION
**Pull request by anthony-chang**
Fixes #4866

cuDF only supports hexadecimal characters up to 0x7f, and we fallback to CPU for characters outside that range. This workaround converts hexadecimal digits directly to the Unicode character itself, allowing GPU support for the full range of hexadecimal digits.

Signed-off-by: Anthony Chang <antchang@nvidia.com>

### CODE REVIEW

**Comment by razajafri** [comment]
Can you add python integration tests?

File: tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionTranspilerSuite.scala [hunk 1]
```diff
@@ -189,6 +181,12 @@ class RegularExpressionTranspilerSuite extends FunSuite with Arm {
     assertCpuGpuMatchesRegexpFind(patterns, Seq("", "\u0007", "a\u0007b", 
         "\u0007\u003f\u007f", "\u007f", "\u007f2"))
   }
+
+  test("hex digits >= 0x7F - find") {
+    val patterns = Seq(raw"\x7f", raw"\x80", raw"\xff", raw"\xfe", raw"\x{7f}", raw"\x{0008f}")
```
Comment (NVnavkumar) [inline tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionTranspilerSuite.scala:186]: Can you add more unit test for larger hex digits (something close to `Character.MAX_CODE_POINT`) using the `\x{...}` syntax?

File: integration_tests/src/main/python/string_test.py [hunk 2]
```diff
@@ -836,6 +836,17 @@ def test_regexp_extract_idx_0():
                 'regexp_extract(a, "^([a-d]*)[0-9]*([a-d]*)\\z", 0)'),
         conf=_regexp_conf)
 
+def test_regexp_hexadecimal_digits():
+    gen = mk_str_gen('[abcd]\\\\x00\\\\x7f\\\\x80\\\\xff[\\\\xa0-\\\\xb0][abcd]')
```
Comment (NVnavkumar) [inline integration_tests/src/main/python/string_test.py:840]: Can you also try regexes with the `\x{...}` syntax in this test?

**Review by NVnavkumar — COMMENTED** [review]
Looks pretty good. Just a couple of minor changes to tests that I would like to see.

File: tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionTranspilerSuite.scala [hunk 3]
```diff
@@ -189,6 +181,12 @@ class RegularExpressionTranspilerSuite extends FunSuite with Arm {
     assertCpuGpuMatchesRegexpFind(patterns, Seq("", "\u0007", "a\u0007b", 
         "\u0007\u003f\u007f", "\u007f", "\u007f2"))
   }
+
+  test("hex digits >= 0x7F - find") {
```
Comment (NVnavkumar) [inline tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionTranspilerSuite.scala:185]: Also you should merge this with the other hex digit to test all of them in the same unit test.

Comment (razajafri) [reply tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionTranspilerSuite.scala:185]: I agree. The scala test can be removed in favor of the python test provided that test covers all cases

**Comment by anthony-chang** [comment]
build

**Review by NVnavkumar — APPROVED** [review]

**Pull request merged**